src/TileManager.js

```diff
@@ -78,13 +78,25 @@ export async function getStorageInfo(urlTemplate) {
  * @param {string} tileUrl
  * @return {Promise<blob>}
  */
-export async function downloadTile(tileUrl) {
-  return fetch(tileUrl).then((response) => {
-    if (!response.ok) {
-      throw new Error(`Request failed with status ${response.statusText}`);
-    }
-    return response.blob();
-  });
+export async function downloadTile(tileUrl,tileKey, alwaysDownload) {
```

Comment:
Thanks for your PR! Please separate responsibility here, checking if the storage has a tile and downloading are two very different things.

Reply:
Also please adapt coding standards of this repo